network-api/networkapi/wagtailpages/tests/research_hub/test_detail_page.py

```diff
@@ -33,6 +36,60 @@ def test_research_detail_page_breadcrumbs(self) -> None:
         self.assertEqual(len(breadcrumbs), 2)
         self.assertEqual(breadcrumbs, expected_breadcrumbs)
 
+    def test_get_research_authors(self) -> None:
+        """
+        This method should return the profiles of all related research authors.
+        """
+        author_profiles = []
+        detail_page = detail_page_factory.ResearchDetailPageFactory(parent=self.library_page, research_authors=[])
+
+        for _ in range(3):
+            research_author_relation = relations_factory.ResearchAuthorRelationFactory(
+                research_detail_page=detail_page
+            )
+            author_profiles.append(research_author_relation.author_profile)
+
+        research_authors = detail_page.get_research_authors()
```

Comment:
Can we [assert the number of queries](https://docs.djangoproject.com/en/4.2/topics/testing/tools/#django.test.TransactionTestCase.assertNumQueries) here, please? We need to make sure that we are not running into a N+1 problem again.

Reply:
Hi @jhonatan-lopes! 

Thanks for working through this with me over zoom, instead of doing the check here, I have added a new function titled "[test_get_research_authors_avoids_n1_queries](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/16637e81be7af07e285942f16607e9e59ef428b2/network-api/networkapi/wagtailpages/tests/research_hub/test_detail_page.py#L63)" as discussed.

It creates 10 related authors, and then checks that the number of queries are less than that.

One concern I do have however and one I was hoping you might be able to help with, is when taking a look at the queries that the assert statement catches (currently catches 2), the queries both seem to be locale related 🤔 


<img width="1138" alt="Screenshot 2023-04-26 at 5 39 27 PM" src="https://user-images.githubusercontent.com/18314510/234731541-16848d9c-7b28-4874-8727-0fa679a35a19.png">

From my understanding, I believe there should be at least 4:
```python
research_author_profiles = localize_queryset(
   Profile.objects.prefetch_related("authored_research").filter(authored_research__research_detail_page=self)
)
```
- 1 for pulling the profiles
- 1 for the `prefetch` statement from the PR feedback
- 1 for `localize_queryset` to get the default locale
- 1 for `localize_queryset` to get the active locale 

I do remember us discussing that the number of queries can vary from time to time. 

Should I wait until we merge in the testing framework from #10536 so I can change the statement to something along the lines of `self.assertNumQueries(< 10)`?

Reply:
Using the debug toolbar on a detail page, I can see that there is a query on profile: 
![image](https://user-images.githubusercontent.com/61277874/234855366-b2cfd98a-65d6-4090-9283-ef3c9c171699.png)

We know the prefetch is working since we can change the N+1 test to run with 100 or 1000 objects and it takes (roughly) the same amount of time. I won't block the PR because of this issue. If you would like to investigate this further, it's up to you. Otherwise, we can remove this test and create a ticket to revisit this another time.

Reply:
Thanks @jhonatan-lopes! 

I have created https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10554 to track that work 👍